shared/utility/motion/InverseKinematics.hpp

```diff
@@ -68,11 +68,16 @@ namespace utility::motion::kinematics {
         const Eigen::Affine3f& leftTarget,
         const Eigen::Affine3f& rightTarget);
 
-    [[nodiscard]] std::vector<std::pair<ServoID, double>> calculateCameraLookJoints(
-        const Eigen::Vector3d& cameraUnitVector);
-
-    [[nodiscard]] std::vector<std::pair<ServoID, float>> calculateHeadJoints(Eigen::Vector3f cameraUnitVector);
-
+    template <typename Scalar>
+    std::vector<std::pair<ServoID, Scalar>> calculateHeadJoints(const Eigen::Matrix<Scalar, 3, 1>& cameraUnitVector) {
+        std::vector<std::pair<ServoID, Scalar>> positions;
+        positions.push_back(std::make_pair(ServoID::HEAD_YAW, atan2(cameraUnitVector.y(), cameraUnitVector.x())));
+        positions.push_back(std::make_pair(ServoID::HEAD_PITCH,
+                                           std::atan2(-cameraUnitVector.z(),
+                                                      std::sqrt(cameraUnitVector.x() * cameraUnitVector.x()
+                                                                + cameraUnitVector.y() * cameraUnitVector.y()))));
+        return positions;
+    }
```

Comment:
This probably breaks formatting and I might have mismatched brackets :p
```suggestion
    [[nodiscard]] std::vector<std::pair<ServoID, Scalar>> calculateHeadJoints(
        const Eigen::Matrix<Scalar, 3, 1>& cameraUnitVector) {
        std::vector<std::pair<ServoID, Scalar>> positions{};
        positions.emplace_back(ServoID::HEAD_YAW, atan2(cameraUnitVector.y(), cameraUnitVector.x()));
        positions.emplace_back(ServoID::HEAD_PITCH,
                               std::atan2(-cameraUnitVector.z(),
                                          std::sqrt(cameraUnitVector.x() * cameraUnitVector.x()
                                                    + cameraUnitVector.y() * cameraUnitVector.y())));
        return positions;
    }
```